src/App.tsx

```diff
@@ -1,39 +1,70 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
+
+  timerIdRename = 0;
+
+  componentDidMount() {
+    this.timerIdRename = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+    document.addEventListener('contextmenu', this.handleHideClock);
+    document.addEventListener('click', this.handleShowClock);
+  }
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
+  componentDidUpdate(...args: [{}, State]) {
+    const { clockName: prevName } = args[1];
+    const { clockName: curName } = this.state;
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    if (prevName !== curName) {
+      // eslint-disable-next-line
+      console.log(`Renamed from ${prevName} to ${curName}`);
+    }
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentWillUnmount() {
+    window.clearInterval(this.timerIdRename);
```

Comment:
you don't unmount your `App` component. Try to kill the renaming timer from the `Clock` component, or add a check, to not rename the clock if `state.hasClock` is `false` 😉